src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperResolutionResultJsonConverter.cs

```diff
@@ -66,7 +67,7 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 
             writer.WriteStartObject();
 
-            writer.WritePropertyArray(nameof(TagHelperResolutionResult.Descriptors), result.Descriptors, _serializer);
+            writer.WritePropertyArray(nameof(TagHelperResolutionResult.Descriptors), result.Descriptors.ToList(), _serializer);
```

Comment:
Same question as above.

Reply:
In this scenario this is right before serialization. I could optimize this better by manually writing out the collection though